guides/release-checklist.md

```diff
@@ -0,0 +1,119 @@
+# Release Checklist for Pulsar
+
+A checklist for releases. This is for the "Regular" releases. (Rolling releases happen automatically.)
+
+## Inputs (Preparation Tasks):
+
+- Candidate commit for the release, with the desired state of the app, on `master` branch (make sure the app is actually ready for release)
+- Changelog up to date with changes since last release
+- Blurb text (some flavor text and info about what happened this release, for GitHub release and blog post)
+
+## Outputs:
+
+- Changelog (within the welcome package + GitHub Release and blog post)
+- Version bump PR
+- GitHub Release
+  - With binaries
+- Update download links on website
+- Blog Post
+- Social posts
+  - Discord announcement
+  - Reddit post
+  - Mastodon post
+
+## Steps:
+
+Note: after the "Changelog" steps are done, the remaining steps can be done in whatever order.
+
+- Prepare the Changelog.
+  - We should make sure last-minute PRs being included in the release are also in the changelog.
+
+- Update the Changelog in-app (in the `welcome` package)
+  - Example PR: https://github.com/pulsar-edit/pulsar/pull/489
+
+
+- Do the version bump pull request.
+  - (Open this once fairly confident the app is ready to roll out, after the in-app Changelog PR is merged -- make sure to include any last-minute changes. If more last-minute changes happen, update the PR and potentially the tag, see next bullet point.)
+  - This should PR be opened from a branch up to date with `master`, plus one commit which changes the version in package.json to the version you're going to release.
+    - Tag that "version update" commit with the tag name starting with v and then the version number of the release. (For example v1.103.0)
```

Comment:
```suggestion
    - Tag that "version update" commit with the tag name starting with v and then the version number of the release. (For example v1.103.0) Like so: `git tag v1.106.0` `git push origin v1.106.0`
```

Do we mind adding something like this? Just in case

Reply:
Wasn't there also a discussion about making sure we had the branch as something like `v1.107.0-release` to avoid conflicts with the tag?

Reply:
@Daeraxa Yes there was, this is a great point, probably something that should be added here as well, as a small aside.

Really things anyone super familiar with git would know, but for everyone else can prove valuable

Reply:
I addressed this line of feedback to be more explicit with git commands to accomplish these steps.

I might have gone overboard, though. And I ended up repeating stuff a bunch of times.

I'm sure this can be streamlined and pared down to be more "checklist" and less "dense field manual". More skimmable and less intimidating to read through, basically. Whilst still being concise and not leaving stuff out.

Anyway, I'm a bit busy today, so this is what I have for now.

Reply:
@DeeDeeG I think this is perfect, personally I don't feel like it's become too dense, since the dense portions are all a few levels deep, with the actual checklist being at the top level essentially. So it's easy to filter mentally once you've read it once, thanks for adding this!